test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+
+            // ###################################################################################################################
+            // input coordinate checks
+            // ###################################################################################################################
+            if (inpCoords == null)
+            {
+                throw new Exception("Null input coordinates.");
+            }
+
+            // check the minimum number of input coordinates for an output geometry
+            switch (outType)
+            {
+                case OgcGeometryType.LineString:
+                    if (inpCoords.Count < 2)
+                        throw new Exception("Input list has fewer than 2 coordinates. To create a polyline, the minimum is 2 coordinates.");
+                    break;
+                case OgcGeometryType.Polygon:
+                    if (inpCoords.Count < 3)
+                        throw new Exception("Input list has fewer than 3 coordinates. To create a polygon, the minimum is 3 coordinates.");
+                    break;
+                default:
+                    throw new Exception("Unsupported output geometry type.");
+            }
+
+            // ###################################################################################################################
+            // check the output geometry type. currently only polygon output is supported. Polylines should use a similar process
+            // with some slight modifications.
+            // ###################################################################################################################
+            switch (outType)
+            {
+                case OgcGeometryType.LineString:
+                    //this will be supported in the future
+                    throw new Exception("Unsupported output geometry type.");
+                case OgcGeometryType.Polygon:
+                    //this is supported
+                    break;
+                default:
+                    // all other output types are not supported
+                    throw new Exception("Unsupported output geometry type.");
+            }
+
+            // ###################################################################################################################
+            // check the input projection info. currently only a geographic input projection is supported. we make many assumptions
+            // based on a geographic coordinate system. if the user wishes to use a projected coordinate system then they will have
+            // to reproject the points first and then use this process.
+            // ###################################################################################################################
+            if (inpProj4Wkt == null || !inpProj4Wkt.StartsWith("GEOGCS"))
```

Comment:
This check (and the parameter) can be removed. One is only attempting to fix dateline issues when one has geographic coordinates. Maybe a test on the extent of the input coordinates as a replacement? In any case, if the width of the envelope is <180, there can't be a dateline crossing.

Reply:
same suggestion I provided, see [reply](https://github.com/NetTopologySuite/NetTopologySuite/pull/580#discussion_r785487001)

Reply:
Users will no doubt try to use it with any projection and realize it doesn't work. I was trying to be proactive about feedback that it is broken when using UTM or other non-geographic projections. But as I said before, what if the UTM projection goes from 50 to 170 meters?